management/connection.go

```diff
@@ -405,6 +406,59 @@ func (c *ConnectionOptionsGoogleOAuth2) SetScopes(enable bool, scopes ...string)
 	tag.SetScopes(c, enable, scopes...)
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+//
+// It is required to handle differences in the allowed_audiences field, which can
+// be an array of strings or a single string.
+func (c *ConnectionOptionsGoogleOAuth2) UnmarshalJSON(data []byte) error {
+	type connectionOptionsGoogleOAuth2 ConnectionOptionsGoogleOAuth2
```

Comment:
Is this type definition necessary?

Reply:
From some investigation it seems that we use this existing pattern because if we use the `ConnectionOptionsGoogleOAuth2` type directly then we will cause an infinite loop as referenced  in [this blogpost](https://jhall.io/posts/go-json-tricks-slightly-custom-marshaler/)

Reply:
Good to know!